### ISSUE TITLE
Question on Prompt Template for Grounding Benchmark

### ISSUE THREAD

**Issue by DEM1TASSE** [yiye3/GUICourse#10]
Hi GUICourse Team,

I’m testing GUICourse on **ScreenSpot** (a grounding benchmark) with this prompt, which is from the paper: `Please give me the grounding box of the "{element_description}".`

I’ve also tried another template: `Task\n{element_description}`, which resulted in lower performance.

Since the model appears to have been trained on different datasets with varying QA templates, I’d like to know if there are any recommended prompts or best practices to achieve optimal results on this benchmark.

Thanks, 
Demi

**Comment by yiye3** [comment]
Thanks for your attention,
You can try thistemplate: `You are a clever and brilliant GUI agent. You can understand the meaning of the image and the text.\n## Your Task\nIf the input is a string, please give me the element regions including this string.\nElse if the input is a region, please give me the string (text) in this region.\n## Input\n<your string or bounding box>\n## Output\n`
Here's an example:
{
              "from": "user",
              "value": "<your image> You are a clever and brilliant GUI agent. You can understand the meaning of the image and the text.\n## Your Task\nIf the input is a string, please give me the element regions including this string.\nElse if the input is a region, please give me the string (text) in this region.\n## Input\nGrrCON 2019 Videos\n## Output\n"
          },
          {
              "from": "assistant",
              "value": "<box>(234,729),(325,748)</box>"
          }

**Comment by DEM1TASSE** [comment]
> Thanks for your attention, You can try thistemplate: `You are a clever and brilliant GUI agent. You can understand the meaning of the image and the text.\n## Your Task\nIf the input is a string, please give me the element regions including this string.\nElse if the input is a region, please give me the string (text) in this region.\n## Input\n<your string or bounding box>\n## Output\n` Here's an example: { "from": "user", "value": " You are a clever and brilliant GUI agent. You can understand the meaning of the image and the text.\n## Your Task\nIf the input is a string, please give me the element regions including this string.\nElse if the input is a region, please give me the string (text) in this region.\n## Input\nGrrCON 2019 Videos\n## Output\n" }, { "from": "assistant", "value": "(234,729),(325,748)" }

Thanks for your repsonse! I'll try it out.

**Comment by DEM1TASSE** [comment]
> Thanks for your attention, You can try this template: `You are a clever and brilliant GUI agent. You can understand the meaning of the image and the text.\n## Your Task\nIf the input is a string, please give me the element regions including this string.\nElse if the input is a region, please give me the string (text) in this region.\n## Input\n<your string or bounding box>\n## Output\n` Here's an example: { "from": "user", "value": " You are a clever and brilliant GUI agent. You can understand the meaning of the image and the text.\n## Your Task\nIf the input is a string, please give me the element regions including this string.\nElse if the input is a region, please give me the string (text) in this region.\n## Input\nGrrCON 2019 Videos\n## Output\n" }, { "from": "assistant", "value": "(234,729),(325,748)" }

Hi, I've tried this prompt, but it seems like the model cannot consistently output bounding boxes.
Here are some examples:

**Input 1:**  
```
You are a clever and brilliant GUI agent. You can understand the meaning of the image and the text.
## Your Task
If the input is a string, please give me the element regions including this string.
Else, if the input is a region, please give me the string (text) in this region.
## Input
Menu button with three horizontal lines located at the top-left corner of the screen.
## Output
```

**Response:**  
```
thoughts: provide an answer based on the image information.
actions: [{"name": "answer", "text": "The menu button with three horizontal lines located at the top-left corner of the screen is Menu."}]
```

---

**Input 2:**  
```
You are a clever and brilliant GUI agent. You can understand the meaning of the image and the text.
## Your Task
If the input is a string, please give me the element regions including this string.
Else, if the input is a region, please give me the string (text) in this region.
## Input
"Restart" option located at the bottom of the power menu in the Start menu.
## Output
```

**Response:**  
```
The "Restart" option located at the bottom of the power menu in the Start menu is in the region with the bounding box coordinates 93.
```
---
And here is my related inference code:
```
def eval_model(args):
    model_path = "local_path_to_qwen_vl_guidance"

    # Load tokenizer
    tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True)

    # Load model with FP16
    model = AutoModelForCausalLM.from_pretrained(
        model_path,
        device_map="auto",
        trust_remote_code=True,
        torch_dtype=torch.bfloat16
    ).eval()

    description = "element_description"
    image_path = "image_path"

    prompt = f"You are a clever and brilliant GUI agent. You can understand the meaning of the image and the text.\n## Your Task\nIf the input is a string, please give me the element regions including this string.\nElse if the input is a region, please give me the string (text) in this region.\n## Input\n{description}\n## Output\n"
            
    print('prompt:', prompt)
    
    query = tokenizer.from_list_format([
        {'image': image_path},
        {'text': prompt},
    ])

    with torch.no_grad():
        response, history = model.chat(tokenizer, query=query, history=None)
    
    print('response:', response)

```
Could you provide further clarification or suggestions on that? Thanks!

**Comment by DEM1TASSE** [comment]
Additionally, here are the test results I have obtained using `qwen_vl_guidance` with prompt: `Please give me the grounding box of the "{element_description}".` on ScreenSpot: 

| Platform      | Accuracy (%) |
|---------------|--------------|
| Mobile Text   | 52.38%       |
| Mobile Icon   | 10.92%       |
| Desktop Text  | 45.88%       |
| Desktop Icon  | 5.71%        |
| Web Text      | 43.04%       |
| Web Icon      | 13.59%       |
| **Average Accuracy** | **28.59%**   |

I would greatly appreciate it if you could let me know if similar tests have been conducted before. Also, I would like to confirm whether these results look correct. 

Thanks!

**Comment by yiye3** [comment]
Oh, we don't enhance Qwen-VL's OCR and text-grounding abilities. Maybe you can try our GUIAgent based on MiniCPM-V. Here is the link: https://huggingface.co/RhapsodyAI/minicpm-guidance/tree/main
